### ISSUE TITLE
DatePicker can throw error when update and disposal happens at the same time

### ISSUE THREAD

**Issue by brunnerh** [carbon-design-system/carbon-components-svelte#878]
Encountered this while using a client-side router (`svelte-spa-router`). The problem appears to be that `onDestroy` is called before `afterUpdate` in some scenarios.

```javascript
// DatePicker.svelte

  afterUpdate(() => {
    if (calendar) {
      if ($range) {
        calendar.setDate([$inputValueFrom, $inputValueTo]);

        // workaround to remove the default range plugin separator "to"
        inputRef.value = $inputValueFrom;
      } else {

        /***** Error here, because `destroy` deletes the `self.config` property, which is used by `setDate` *****/
        calendar.setDate($inputValue);
      }
    }
  });

  onDestroy(() => {
    if (calendar) {
      calendar.destroy();
    }
  });
```

I would suggest that `calendar` is set to `null` after `destroy()`, so access in `afterUpdate` is prevented by the existing `if (calendar)` statement. (Tested this by patching the component locally and it solved the issue.)

Could not reproduce this in the REPL without the router and the router cannot be used in the REPL.
Here is a REPL-based repository exhibiting the error: https://github.com/brunnerh/date-picker-issue

**Comment by metonym** [comment]
Nice catch – thank you for providing a repro.

I agree – setting `calendar` to `null` would be the best approach.

**Comment by metonym** [comment]
Fixed in [v0.48.0](https://github.com/carbon-design-system/carbon-components-svelte/releases/tag/v0.48.0)
